### DISCUSSION TITLE
Computational complexity

### DISCUSSION THREAD

**Discussion by rnd0101** [orcmid/miser#54] [Q&A]
[Models of computation](https://en.wikipedia.org/wiki/Model_of_computation) usually go hand-in-hand with computational complexity "payloads". This is probably more into engineering side than pure platonic point of view, but it already became common practice to provide cost of computation for models.

I believe complexity of the implementation is only partially depending on the "engineering" aspects, while another part is complexity of the theory itself, inherent complexity. This statement is justified by notion that both theory and implementation are **systems**. And domain of complexity notion's application are systems (in almost any definition).

(disclaimer: I am trying to enter my favorite philosophical area, which is Complexity.)

Offtopic (?). While speaking of systems, I recalled the best formalism about complex system made by soviet/ukrainian late philosopher Avenir Uyemov (actually, he had a whole philosophical school in the days of USSR). Unfortunately, he wrote very little in English, but I've found one of his [works](http://www.philosof.onu.edu.ua/elb/uemov/ternar_3.pdf)  by googling. His ternary language due to it's utter genericity covers any systems nicely. So far, I have not found any better coverage in Western philosophy (please, point me if you know). (this is "Plato") And then there is "engineering" / design "Nerd" approach to systems, which experienced quite a hype in Western countries recently, while in it's origin it seems to be being forgotten: [TRIZ](https://en.wikipedia.org/wiki/TRIZ) (it's being blamed to not being scientific (eg, due to some metaphors, which remind physical phenomena, but are not), but I argue - it should not be! It's about systematic and rational approach design and engineering versus Trial-n-Error.)

**Comment by orcmid** [comment]
* It is interesting that you consider conceptual complexity, rather different than computational complexity as commonly used.

 * Thank you for the link to [TRIZ](https://en.wikipedia.org/wiki/TRIZ).  I [am a fan](http://trosting.org/info/2005/08/i050803b.htm) of [Pattern Languages](https://en.wikipedia.org/wiki/TRIZ) though not so much the derivative form that has arisen in software development.
